test/quantization/fx/test_equalize_fx.py

```diff
@@ -110,3 +124,142 @@ def test_input_weight_observer(self, input_qdtype, input_qscheme, weight_qdtype,
+        new_input_equalization_observer = _InputEqualizationObserver.with_args(
+            dtype=torch.qint8, qscheme=torch.per_tensor_symmetric, output_obs=output_obs)
+        equalization_qconfig = QConfig(activation=new_input_equalization_observer,
+                                       weight=weight_equalization_observer)
+
+    def test_input_weight_equalization_prepare_linear_conv(self):
+        """ Tests that observers are inserted correctly for a linear then conv layer
+        """
+        class M(nn.Module):
+            def __init__(self):
+                super(M, self).__init__()
+                self.linear = nn.Linear(5, 5)
+                self.conv = nn.Conv2d(3, 3, 1, 1)
+
+            def forward(self, x):
+                x = self.linear(x)
+                x = self.conv(x)
+                return x
+
+        m = M().eval()
+
+        # Test default case: should insert a MinMaxObserver as an output observer
+        qconfig_dict = {
+            "": default_qconfig,
+            "object_type": [(nn.Linear, default_equalization_qconfig)]}
+        prepared = prepare_fx(m, qconfig_dict)
+
+        node_occurrence = {
+            ns.call_module(_InputEqualizationObserver): 1,
+            ns.call_module(MinMaxObserver): 2,
+        }
+        self.checkGraphModuleNodes(prepared, expected_node_occurrence=node_occurrence)
+
+    def test_input_weight_equalization_prepare_functional_linear(self):
+        """ Tests that observers are inserted correctly for two functional layers
+        """
+        class Linear(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.w = torch.ones(5, 5)
+                self.b = torch.zeros(5)
+
+            def forward(self, x):
+                return nn.functional.linear(x, self.w, self.b)
+
+        class M(torch.nn.Module):
+            def __init__(self):
+                super().__init__()
+                self.linear1 = Linear()
+                self.linear2 = Linear()
+
+            def forward(self, x):
+                x = self.linear1(x)
+                x = self.linear2(x)
+                return x
+
+        m = M().eval()
+
+        # Test default case: should insert a MinMaxObserver as an output observer
+        qconfig_dict = {
+            "": default_qconfig,
+            "object_type": [(Linear, default_equalization_qconfig)]}
+        prepared = prepare_fx(m, qconfig_dict)
+
+        node_occurrence = {
+            ns.call_module(_InputEqualizationObserver): 1,
+            ns.call_module(_WeightEqualizationObserver): 2,
+            ns.call_module(MinMaxObserver): 2,
+        }
+        self.checkGraphModuleNodes(prepared, expected_node_occurrence=node_occurrence)
+
+    def test_input_weight_equalization_prepare_functional_linear_op(self):
+        """ Tests that observers are inserted correctly for two functional
+        linear layers with a fp32 operation in between
```

Comment:
The mul operator in between is still quantized (you can call convert_fx to confirm this) because you specified the default qconfig for all operators. To simulate an fp32 operator you could add, say a conv in between two linear layers and set the qconfig dict to default to None for all other ops `qconfig_dict = { "" : None, ...}`